### ISSUE TITLE
Add support for TextToSpeech WebSockets

### ISSUE THREAD

**Issue by StephenHodgson** [RageAgainstThePixel/ElevenLabs-DotNet#46]
[Support websockets for text to speech](https://elevenlabs.io/docs/api-reference/websockets)

ElevenLabs-DotNet-Proxy should also support forwarding websockets connections

**Comment by ocinon** [comment]
@StephenHodgson did you start implementing WebSockets by any chance?
Also, I saw the speech-to-speech model in your 3.0.0 draft, but there is no support yet, correct?

**Comment by StephenHodgson** [comment]
Yes I was already doing this for the unity package and was considering porting it once done

**Comment by ocinon** [comment]
@StephenHodgson I couldn't find any previous WebSocket implementation in your Unity repo. As I needed it, I implemented it for the DotNet version here:
ocinon/ElevenLabs-DotNet@93457e124ed0397bf3532c6fd2b62c9188406d41

It extends the client slightly and tries to pick up the same patterns the repo used before. It lacks proxy support and tests. If you have any notes, let me know.

**Comment by StephenHodgson** [comment]
> @StephenHodgson I couldn't find any previous WebSocket implementation in your Unity repo. As I needed it, I implemented it for the DotNet version here: [ocinon/ElevenLabs-DotNet@93457e1](https://github.com/ocinon/ElevenLabs-DotNet/commit/93457e124ed0397bf3532c6fd2b62c9188406d41)
> 
> It extends the client slightly and tries to pick up the same patterns the repo used before. It lacks proxy support and tests. If you have any notes, let me know.

Feel free to open a pull request!

**Comment by odillner** [comment]
Any updates on this? It would be very useful in a project I'm part of.

**Comment by ocinon** [comment]
Sorry for never updating the thread. After some back-and-forth with ElevenLabs support, it turned out that their WebSocket implementation has a 20-second timeout. This is fine for batch conversions but makes it pretty useless for low-volume or prototyping voice-to-voice bots or similar use cases.

It might be possible to keep sending a space string (" ") as a keep-alive signal, but I stopped spending more time on it, as during testing, I didn't get speed increases compared to the REST API (but I didn't do proper testing). The code exists, and I could push it for reference.

**Comment by odillner** [comment]
Thanks for the quick response!

Well that's disappointing, but thanks for doing the legwork. 

I'm gonna do some testing on my own, so please push the code.

**Comment by ocinon** [comment]
It's here [ocinon/ElevenLabs-DotNet](https://github.com/ocinon/ElevenLabs-DotNet/tree/development)

I updated it to the latest ElevenLabs version. Keep-alive messages don't seem to work.

Some basic testing code:

```c#
using ElevenLabsClient client = new(ELEVEN_LABS_KEY);
await using FileStream fileStream
	= new("output.mp3", FileMode.Create, FileAccess.Write, FileShare.Read);
await client.TextToSpeechWebSocketEndpoint.StartTextToSpeechAsync(
	Voice.Arnold, (async voiceClip =>
		              {
			              if (voiceClip == null)
			              {
				              Console.WriteLine("Received null voice clip.");
				              return;
			              }

			              Console.WriteLine(
				              $"Received voice clip with {voiceClip.ClipData.Length} bytes.");
			              await fileStream.WriteAsync(voiceClip.ClipData);
		              })
);
while (true)
{
	Console.Write("Enter text to convert to speech: ");
	string? text = Console.ReadLine();
	if (text is null) { continue; }

	if (text == "exit") { break; }

	switch (text)
	{
		case "flush":
		{
			await client.TextToSpeechWebSocketEndpoint.SendTextToSpeechAsync(".", true, false);
			continue;
		}
		case "trigger":
		{
			await client.TextToSpeechWebSocketEndpoint.SendTextToSpeechAsync(".", null, true);
			continue;
		}
		default:
		{
			await client.TextToSpeechWebSocketEndpoint.SendTextToSpeechAsync(text);
			break;
		}
	}
}

await client.TextToSpeechWebSocketEndpoint.EndTextToSpeechAsync();
```

**Comment by StephenHodgson** [comment]
@ocinon feel free to open a PR on the main project for everyone else to get :)

**Comment by ocinon** [comment]
@StephenHodgson should we push it into the development branch for now? Could you open that one for me?

**Comment by StephenHodgson** [comment]
Sure I'll push a development branch right now for you to target :)

**Comment by StephenHodgson** [comment]
you may want to rebase your changes tho and just make sure you've synced with upstream.

**Comment by ocinon** [comment]
It's up to date but not rebased. One sec.

**Comment by ocinon** [comment]
Done
